<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>968</width>
    <height>771</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Thermal &amp; Mechanical Control System</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="3,1">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout" stretch="2,2,2,1">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="PlotWidget" name="force_plot_widget">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="disp_plot_widget">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="temp_plot_widget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>2</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="pwm_plot_widget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>1</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,2,2">
        <property name="spacing">
         <number>20</number>
        </property>
        <item>
         <widget class="QGroupBox" name="grp_status">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="title">
           <string>Current Status</string>
          </property>
          <layout class="QGridLayout" name="layout_status">
           <property name="sizeConstraint">
            <enum>QLayout::SetMaximumSize</enum>
           </property>
           <item row="0" column="1">
            <widget class="QLabel" name="lbl_temp">
             <property name="font">
              <font>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>0.0 °C</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <widget class="QLabel" name="lbl_pwm">
             <property name="font">
              <font>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>PWM: 0%</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QLabel" name="lbl_fan">
             <property name="font">
              <font>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>FAN: OFF</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="lbl_static_temp">
             <property name="font">
              <font>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>Temperature:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="grp_control">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="title">
           <string>Control Settings</string>
          </property>
          <layout class="QVBoxLayout" name="layout_control">
           <property name="sizeConstraint">
            <enum>QLayout::SetMaximumSize</enum>
           </property>
           <item>
            <widget class="QRadioButton" name="radio_manual">
             <property name="font">
              <font>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>Manual Mode</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radio_pid">
             <property name="font">
              <font>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>PID Mode</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="widget_manual" native="true">
             <layout class="QHBoxLayout" name="lay_man">
              <item>
               <widget class="QLabel" name="lbl_pwm_in">
                <property name="text">
                 <string>PWM (%):</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="spin_pwm">
                <property name="font">
                 <font>
                  <weight>50</weight>
                  <bold>false</bold>
                 </font>
                </property>
                <property name="maximum">
                 <number>100</number>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QCheckBox" name="chk_fan">
                <property name="text">
                 <string>FAN ON</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="widget_pid" native="true">
             <layout class="QHBoxLayout" name="lay_pid">
              <item>
               <widget class="QLabel" name="lbl_target">
                <property name="text">
                 <string>Target Temp:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="spin_target">
                <property name="font">
                 <font>
                  <weight>50</weight>
                  <bold>false</bold>
                 </font>
                </property>
                <property name="value">
                 <double>25.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_apply">
             <property name="font">
              <font>
               <weight>50</weight>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>Apply Settings</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_start">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
    background-color: #2ECC71; /* 배경색 (초록색) */
    color: white;              /* 글자색 (흰색) */
    border-radius: 5px;        /* 모서리 둥글게 */
    font-weight: bold;         /* 글자 진하게 */
    padding: 5px;
}

/* 마우스를 올렸을 때 색상 변화 */
QPushButton:hover {
    background-color: #27AE60;
}

/* 클릭했을 때 색상 변화 */
QPushButton:pressed {
    background-color: #1E8449;
}</string>
             </property>
             <property name="text">
              <string>START SYSTEM</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_stop">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
    background-color: #E74C3C; /* 배경색 (빨간색) */
    color: white;              /* 글자색 (흰색) */
    border-radius: 5px;        /* 모서리 둥글게 */
    font-weight: bold;         /* 글자 진하게 */
    padding: 5px;
}

/* 마우스를 올렸을 때 색상 변화 */
QPushButton:hover {
    background-color: #C0392B;
}

/* 클릭했을 때 색상 변화 */
QPushButton:pressed {
    background-color: #A93226;
}</string>
             </property>
             <property name="text">
              <string>STOP (Emergency)</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="grp_save">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="title">
           <string>Data Recording</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_save">
           <property name="sizeConstraint">
            <enum>QLayout::SetMaximumSize</enum>
           </property>
           <item>
            <widget class="QPushButton" name="btn_log_start">
             <property name="styleSheet">
              <string notr="true">QPushButton {
    background-color: #2ECC71; /* 배경색 (초록색) */
    color: white;              /* 글자색 (흰색) */
    border-radius: 5px;        /* 모서리 둥글게 */
    font-weight: bold;         /* 글자 진하게 */
    padding: 5px;
}

/* 마우스를 올렸을 때 색상 변화 */
QPushButton:hover {
    background-color: #27AE60;
}

/* 클릭했을 때 색상 변화 */
QPushButton:pressed {
    background-color: #1E8449;
}</string>
             </property>
             <property name="text">
              <string>START LOGGING</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_log_stop">
             <property name="styleSheet">
              <string notr="true">QPushButton {
    background-color: #E74C3C; /* 배경색 (빨간색) */
    color: white;              /* 글자색 (흰색) */
    border-radius: 5px;        /* 모서리 둥글게 */
    font-weight: bold;         /* 글자 진하게 */
    padding: 5px;
}

/* 마우스를 올렸을 때 색상 변화 */
QPushButton:hover {
    background-color: #C0392B;
}

/* 클릭했을 때 색상 변화 */
QPushButton:pressed {
    background-color: #A93226;
}</string>
             </property>
             <property name="text">
              <string>STOP LOGGING</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_log_reset">
             <property name="styleSheet">
              <string notr="true">/* Reset 버튼: 어두운 노란색 (Amber) */
QPushButton#btn_log_reset {
    background-color: #FFA000;  /* 진한 노란색/호박색 */
    color: #000000;             /* 가독성을 위해 글자는 검은색 */
    border-radius: 4px;
    font-weight: bold;
    padding: 5px;
    border: 1px solid #E69100;
}

QPushButton#btn_log_reset:hover {
    background-color: #FFB300;  /* 마우스 올리면 살짝 밝아짐 */
}

QPushButton#btn_log_reset:pressed {
    background-color: #CC8400;  /* 클릭 시 더 어두워짐 */
}

/* 버튼이 비활성화(Disabled) 되었을 때 */
QPushButton#btn_log_reset:disabled {
    background-color: #BDBDBD;
    color: #757575;
}</string>
             </property>
             <property name="text">
              <string>RESET </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_fname">
             <property name="text">
              <string>File Name:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="edit_filename">
             <property name="placeholderText">
              <string>Empty = Timestamp</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btn_save">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>60</height>
              </size>
             </property>
             <property name="styleSheet">
              <string>background-color: #FF9800; color: white; font-weight: bold; border-radius: 5px;</string>
             </property>
             <property name="text">
              <string>SAVE DATA
(CSV + PNG)</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
